roles/ceph-config/templates/ceph.conf.j2

```diff
@@ -121,7 +121,7 @@ osd memory target = {{ _osd_memory_target | default(osd_memory_target) }}
 host = {{ _rgw_hostname }}
 keyring = /var/lib/ceph/radosgw/{{ cluster }}-rgw.{{ _rgw_hostname + '.' + instance['instance_name'] }}/keyring
 log file = /var/log/ceph/{{ cluster }}-rgw-{{ hostvars[host]['ansible_hostname'] + '.' + instance['instance_name'] }}.log
-rgw frontends = {{ radosgw_frontend_type }} {{ 'port' if radosgw_frontend_type == 'civetweb' else 'endpoint' }}={{ instance['radosgw_address'] }}:{{ instance['radosgw_frontend_port'] }} {{ radosgw_frontend_options }}
+rgw frontends = {{ radosgw_frontend_type }} {{ 'port' if radosgw_frontend_type == 'civetweb' else 'endpoint' }}={{ instance['radosgw_address'] }}:{{ instance['radosgw_frontend_port'] }}{{ 's ssl_certificate='+radosgw_civetweb_ssl_certificate if radosgw_civetweb_ssl_certificate }} {{ radosgw_frontend_options }}
```

Comment:
This doesn't work becaise you need a else statement

```console
AnsibleUndefinedVariable: the inline if-expression on line 1 evaluated to false and no else section was defined.
```
Alternatively you could use something like :

```yaml
{% if radosgw_civetweb_ssl_certificate %}s ssl_certificate={{ radosgw_civetweb_ssl_certificate }}{% endif %}
```

But the real issue there is that this code will be also true for beast frontend if `radosgw_civetweb_ssl_certificate` is set but the ssl parameters are different.

Reply:
maybe we should use a bigger condition and set the parameters based on _ssl_certificate and _frontend_type ?

Reply:
updated with fixes as per comments and to deal better with both civetweb and beast